frontend/src/global/form-rules.ts

```diff
@@ -806,4 +820,9 @@ export const Rules: CommonRule = {
         validator: checkIpv4,
         trigger: 'blur',
     },
+    phone: {
+        validator: checkPhone,
+        required: true,
+        trigger: 'blur',
+    },
 };
```

Comment:
The provided code checks for common rules like Https checking and phone validation before returning the `commonRules` object with specific validators.

Potential Issues:

1. Regular expression mismatch: The current regular expression is designed to match all digits including those that might also be followed by an extension but we're trying to check just numbers here which causes some mismatches.
2. Input Validation missing logic: This part seems to validate only input text type field without performing actual input processing.

Solution Suggestions:

- Validate entire string not just regex matches: In the first condition of `if`, change it so this validates the whole value instead of simply matching against a regex pattern.

Optimization Suggestions:

1. Consolidate repetitive validations into single line functions to improve readability
      ```js
       const isNumberCheck = (input, msg) => {
           if (!/^[+-]?(\d+(\.\d*)?$)(%)？]{0,}$/.test(input)) {
               return new ValidationError(msg)
          };
```

2. Use functional programming style instead of imperative loops when implementing rule implementations:
   ```javascript

        const getValidatorValueByField = (fieldName, fieldValue, formItem) =>
            fieldValue ? { [fieldName]: fieldValue.toString() } : {};

        const getValueFromValidatedFormItems = async (validatorResults, formItemRulesMap) => {
             let resFields = {};
                Object.keys(validatorResults).forEach((key) => {
                    if(Object.prototype.hasOwnProperty.call(formItemRulesMap , key)){
                        const {fieldConfig} = formItemRulesMap[key];
                       // use 'getValuFromValidatedFormItems' function from above to create mapped values for each field
                        resFields[fieldConfig.propertyKey] =(await getValuesForPropRes(resFields, getFieldByName(fieldConfig.propName))) ;
                            await addValidationMessagesToFieldIfNecessary(fieldConfig);
                    
                    }
                 });
                   return {...resFields};
             
```